executor/services/agent_service.py

```
@@ -315,89 +315,12 @@ async def send_cancel_callback_async(self, task_id: int) -> None:
-            logger.info(
-                f"[{_format_task_log(task_id, subtask_id)}] Cleaned up task state after cancel"
-            )
-
-            # Close agent client and remove session to free up slot
-            # This ensures the task_id is no longer reported in heartbeat
-            await self._cleanup_cancelled_session(task_id, session)
+            # DO NOT cleanup task state here - SDK interrupt messages still need to be processed
+            # State will be cleaned up in response_processor after all messages are processed
 
         except Exception as e:
             logger.exception(f"[{task_id}] Error sending cancel callback: {e}")
```

Ruff (0.14.14)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)